chia/data_layer/download_data.py

```diff
@@ -283,21 +288,25 @@ async def insert_from_delta_file(
                 existing_generation,
                 group_files_by_store,
             )
-            await insert_into_data_store_from_file(
+            num_inserted = await insert_into_data_store_from_file(
                 data_store,
                 store_id,
                 None if root_hash == bytes32([0] * 32) else root_hash,
                 target_filename_path,
             )
             log.info(
                 f"Successfully inserted hash {root_hash} from delta file. "
-                f"Generation: {existing_generation}. Store id: {store_id}."
+                f"Generation: {existing_generation}. Store id: {store_id}. Nodes inserted: {num_inserted}."
             )
 
-            root = await data_store.get_tree_root(store_id=store_id)
-            with open(filename_full_tree, "wb") as writer:
-                await data_store.write_tree_to_file(root, root_hash, store_id, False, writer)
-            log.info(f"Successfully written full tree filename {filename_full_tree}.")
+            if target_generation - existing_generation <= maximum_full_file_count - 1:
```

Comment:
maybe?  or just mark this conversation as resolved.

```suggestion
            if target_generation - existing_generation < maximum_full_file_count:
```